File: src/main/java/ru/citeck/ecos/records2/source/dao/RecordsQueryDAO.java
package ru.citeck.ecos.records2.source.dao;

import ru.citeck.ecos.records2.RecordRef;
import ru.citeck.ecos.records2.request.query.RecordsQuery;
import ru.citeck.ecos.records2.request.query.RecordsQueryResult;

public interface RecordsQueryDAO extends RecordsQueryBaseDAO {

    RecordsQueryResult<RecordRef> queryRecords(RecordsQuery query);
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/dao/query/RecordsQueryDao.kt
package ru.citeck.ecos.records3.record.dao.query

import ru.citeck.ecos.records3.record.dao.RecordsDao
import ru.citeck.ecos.records3.record.dao.query.dto.query.RecordsQuery
import kotlin.jvm.Throws

/**
 * You can return:
 *
 * List<*>
 * Set<*>
 * RecsQueryRes<*>
 * DataValue with array
 *
 * if none of any types above returned, then listOf(result) will be returned from DAO
 */
interface RecordsQueryDao : RecordsDao {

    @Throws(Exception::class)
    fun queryRecords(recsQuery: RecordsQuery): Any?
}
